cdisc_rules_engine/operations/parent_library_model_column_order.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+from collections import OrderedDict
+from cdisc_rules_engine.operations.library_model_column_order import (
+    LibraryModelColumnOrder,
+)
+from pandas import Series
+
+
+class ParentLibraryModelColumnOrder(LibraryModelColumnOrder):
+    def _execute_operation(self):
+        """
+        Fetches column order for a supp's parent domain from the CDISC library.
+        Returns it as a Series of lists like:
+        0    ["STUDYID", "DOMAIN", ...]
+        1    ["STUDYID", "DOMAIN", ...]
+        2    ["STUDYID", "DOMAIN", ...]
+        ...
+
+        Length of Series is equal to the length of given dataframe.
+        The lists with column names are sorted
+        in accordance to "ordinal" key of library metadata.
+        """
+        if "RDOMAIN" not in self.params.dataframe:
+            return []
+        rdomains = self.params.dataframe["RDOMAIN"].unique()
+        rdomain_names_list = {}
+        for rdomain in rdomains:
+            parent_datasets = [
```

Comment:
Instead of having to loop over the datasets for each rdomain, can't you get the list of datasets corresponding to an rdomain by looping over the datasets and generating a map of rdomain -> parent datasets. Then for each value in the map you can get the appropriate metadata?

Reply:
I hadn't thought about this. Since this operation is primarily for supps, which should always have the same rdomain throughout, should I just take the first RDOMAIN value instead of generating variable mappings for all parent domains?

Reply:
If you think this is an appropriate assumption to make, but I don't think we need to make that assumption

Reply:
I switched to setdefault to avoid the extra loop through the records and to avoid getting variable lists for unused domains

Reply:
Ah, I thought your concern was with looping through the dataset rows an extra time here: `self.params.dataframe["RDOMAIN"].unique()`

I don't think we want to reduce the loops through the dataset list at the cost of doing unnecessary metadata lookups. Typically this Operation would involve:
`1 single rdomain` * `a study of ~25 datasets.`
That's small. I could create a dict of domains -> datasets, which would be useful to have globally across the entire validation, but the gains here would still be marginal

Reply:
My concern is this loop here: https://github.com/cdisc-org/cdisc-rules-engine/pull/337/files#diff-91c2b79e0fca32264501ab5b7b25327d9c1a67f387c70b3127c95db0703502daR32 If we call it for each RDOMAIN we'd have to loop through datasets per rdomain. I guess we can assume there will usually only be 1 rdomain, but we can avoid running that loop all together by just building a map domains -> datasets. While it would be a slight gain overall, I think that map would be particularly useful here to avoid relying on the assumption that there is only 1 single rdomain in the dataset